### PR TITLE
Allow operators to opt into publishing node and alloc metrics

### LINKED ISSUE #1499: Filter allocation metrics from statsd metrics
We would like to have a parameter to toggle off the live allocation metrics and just emit node level metrics to statsd.

Our use case for this is to leverage librato for capacity planning and sending all allocation metrics to librato is costly.

In the future we may like to target specific allocations to emit allocation level metrics out of on a on-demand basis but we could use the command line for those specific cases.

### PR DESCRIPTION
**Pull request by diptanu**
Fixes #1499

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -1397,7 +1397,11 @@ func (c *Client) collectHostStats() {
 			c.resourceUsageLock.Lock()
 			c.resourceUsage = ru
 			c.resourceUsageLock.Unlock()
-			c.emitStats(ru)
+
+			// Publish Node metrics if operator has opted in
+			if c.config.PublishNodeMetrics {
```
Comment (slackpad) [inline client/client.go]: It looks like you check the config settings inside the function. Are you sure you need the guard up here?

Comment (diptanu) [inline client/client.go]: @slackpad We do that in the `emitMetrics` method of task runners, this is for the `emitMetrics` method of the Client which sends host related metrics.

**Comment by diptanu** [comment]
cc/ @dadgar

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 ## 0.4.1 (UNRELEASED)
 
+__BACKWARDS INCOMPATIBILITIES:__
```
Comment (dadgar) [inline CHANGELOG.md]: I would make it default as true and let people who want to disable it disable it.

Comment (diptanu) [inline CHANGELOG.md]: We are keeping the default false since the volume of metrics Nomad emits is quite large.

**Comment by diptanu** [comment]
We will open a follow up PR which will allow a tunable for operators to opt-in for nomad-internals related metrics.

**Pull request merged**